extensions/2.0/Khronos/KHR_animation2/README.md

```diff
@@ -0,0 +1,187 @@
+# KHR\_animation2
+
+## Contributors
+
+* Alexey Knyazev
+* Ben Houston, Threekit
+* Bryce Hutchings, Microsoft [@brycehutchings](https://twitter.com/brycehutchings)
+* Don McCurdy
+* Gary Hsu, Microsoft [@bghgary](https://twitter.com/bghgary)
+* Jamie Marconi, Microsoft [@najadojo](https://twitter.com/najadojo)
+* Joe Herdman, [@jbherdman](https://github.com/jbherdman)
+* Lewis Weaver, Microsoft
+* Norbert Nopper, UX3D [@UX3DGpuSoftware](https://twitter.com/UX3DGpuSoftware)
+
+Copyright (C) 2018-2021 The Khronos Group Inc. All Rights Reserved. glTF is a trademark of The Khronos Group Inc.
+See [Appendix](#appendix-full-khronos-copyright-statement) for full Khronos Copyright Statement.
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension is based on the animation features of glTF 2.0. The structure of the schemas stay the same.  
+  
+The only major addition is, that the output values are mapped using a [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901).
+  
+### Motivation
+
+At this point of time, one can only target the transformation or weight data of a node.  
+With this extension, one can technically target any value (`scalar`, `vec2`, `vec3`, `vec4` and scalars in an `array`) in glTF e.g.
+
+* Color factors in materials
+* Camera field of view
+
+Even using a JSON pointer, the targets and their expected behavior are explictly defined.
+
+Current and future extensions have to write against this specification to allow to animate specific values.
+
+#### Current and future restrictions
+
+The calculated values can be out of range of the minimum and maximum value, however the values have to be clamped using the properties min and max value before setting the value.
```

Comment:
I'm not sure I understand this part – should min and max values reflect the full range of new animation data? (I think yes) - why can they be out of range then?

Reply:
I would also clarify whether "before writing back" and "before setting the value" are referring to authoring tool or client behavior. I think you mean client behavior?

Not objecting to that suggestion in the extension if it's generally preferred, but we realistically can't/won't implement this clamping in three.js. Interpolators are optimized mathematical functions, not containers of state tied to a particular file format. If three.js itself supports a value (e.g. `emissiveFactor = [5.0, 0.0, 0.0]`) I can't see us imposing an additional clamp when the material or animation came from glTF as opposed to another format.

Reply:
Yes, I was meaning client behaviour. In your case, you are moving the problem to the authoring tool. In some cases, it has to sample the data.
Furthermore, this approach is from FBX.
Hmm, maybe we add this feature to the animation itself? That there explicitly one can define, that the animation is clamped in a specific range.

Reply:
We are just talking about cases where CUBICSPLINE in/out tangents might send a value outside the "intended" range, correct? I don't think I can see why this is more of an issue for bounded values like color than for unbounding values like scale or morph weights – it's still the artist and/or authoring tool's responsibility to set the curves. 

Would it be enough to say, something like:

> Keyframe values are limited to the defined range of the animated property. When interpolation yields results individual frame values outside this range, results are undefined.